src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -98,239 +102,151 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes)
             // or keep the dir/file and write proper cleanup when application closes
         }
 
-        public List<Tensor> Run(List<Tensor> inputTensors)
+        /// <summary>
+        /// Uses an open session to score a list of NamedOnnxValues.
+        /// </summary>
+        /// <param name="inputNamedOnnxValues">The NamedOnnxValues to score</param>
+        /// <returns>Resulting output NamedOnnxValues list</returns>
+        public List<NamedOnnxValue> Run(List<NamedOnnxValue> inputNamedOnnxValues)
         {
-            var outputTensors = _modelManager.RunModel(
-                _modelName, _ignoredVersion, InputNames, inputTensors, OutputNames);
-
-            return outputTensors;
+            var outputNameOnnxValues = _session.Run(inputNamedOnnxValues);
+            var results = new List<NamedOnnxValue>();
+            foreach (var value in outputNameOnnxValues)
+            {
+                results.Add(value);
+            }
+            return results;
         }
 
+        /// <summary>
+        /// Convert the model to a byte array.
+        /// </summary>
+        /// <returns>byte[]</returns>
         public byte[] ToByteArray()
         {
             return File.ReadAllBytes(_modelFile);
         }
 
-        private OnnxNodeInfo[] GetInputsInfo()
```

Comment:
>private [](start = 8, length = 7)

why not keep private? these functions are private utilities. ModelInfo property already contains this information for use by other classes.

Reply:
made private

Reply:
Marked it as private.